specs/config-standard.md

```diff
@@ -0,0 +1,43 @@
+# Configuration Standard
+
+## Purpose
+
+Define how configuration should be implemented across all services within Astria services.
+
+## Background
+
+When running a service, it is generally [best practice to store configuration in environment variables](https://12factor.net/config), without defaults. This makes it difficult to commit environment specific config, and ensures separation of configuration and code. It makes it easier to integrate with KMS for secrets, and container orchestration systems are configured to enable this easily. Not having defaults ensures configuration is done with intention on a given environment.
+
+For local development, however, this is often inconvenient. CLI args and defaults are often easier to manage while running on your local machine. Additionally, blockchains often have genesis files which need to be coordinated across a diverse set of decentralized actors. Having files for this type of configuration is often best, and the configurations may be managed via command line tools.
+
+This standard is intended to balance the best practice, with the needs of developers for ease, and some shared configuration.
+
+## Astria Standard
+
+> Note that this standard is not implemented across repo, there are tagged issues to update services to use the standard [here](https://github.com/astriaorg/astria/issues/240).
+
+- Configuration in Rust managed via [Figment](https://docs.rs/figment/latest/figment/)
```

Comment:
Let's not enforce the use of a particular crate - to make reporting better we should probably be move away from figment and roll our own library on top of serde because a) we don't use most of figment, b) we have the issue of short circuiting if a config is missing so that only one gets reported, and c) the log only contains the config field but not the full name of the missing env var.

So for example if we can't find `ASTRIA_SERVICE_FOO` and `ASTRIA_SERVICE_BAR`, we are getting the very unhelpful message that `foo` is missing.

Reply:
We can update the spec when we change that. I'd like to have some standardization for now, just an established pattern this will make the rust code very similar across binaries. When we change, we can update with an established easy to swap everything pattern as well.